x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_actions/rule_params_modifier.ts

```diff
@@ -58,6 +59,10 @@ const applyBulkActionEditToRuleParams = (
         "Index patterns can't be overwritten. Machine learning rule doesn't have index patterns property"
       );
 
+      if (action.overwriteDataViews) {
+        ruleParams.dataViewId = undefined;
+      }
```

Comment:
@dhurley14 I believe, overwrite should also be applied on set/overwrite index patterns, as data view checkbox is also available for it
<img width="394" alt="Screenshot 2022-07-29 at 16 18 03" src="https://user-images.githubusercontent.com/92328789/181791277-06b20620-6408-47d3-b24f-58b52e7d2e9a.png">